src/Symfony/Bundle/WebProfilerBundle/EventListener/TurboDriveCspListener.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\Attribute\When;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Show the Web Debug Toolbar (WDT) when navigating via Turbo Drive and when a strict Content-Security-Policy is set.
+ * This is done by reusing WDT's nonces.
+ * This service is registered only when TurboBundle is installed.
+ *
+ * @author David Petrásek <davidpetrasek@hotmail.cz>
+ */
+class TurboDriveCspListener implements EventSubscriberInterface
+{
+    public function onKernelResponse(ResponseEvent $event): void
+    {
+        $response = $event->getResponse();
+        $responseContent = $response->getContent();
+
+        if (!str_contains($responseContent, '<div id="sfwdt')) {
+            return;
+        }
+
+        $csp = $response->headers->get('Content-Security-Policy');
+        if (!$csp) {
+            return;
+        }
+
+        $scriptContent = <<<'EOD'
+            document.addEventListener('turbo:before-fetch-request', (event) =>
+            {
+                var wdt = document.querySelector('.sf-toolbar');
+                if (wdt)
+                {
+                    let wdtStyle = wdt.nextElementSibling;
+                    let wdtScript = wdtStyle.nextElementSibling;
+
+                    if (wdtStyle.nonce) {event.detail.fetchOptions.headers['X-SymfonyProfiler-Style-Nonce'] = wdtStyle.nonce;}
+                    if (wdtScript.nonce) {event.detail.fetchOptions.headers['X-SymfonyProfiler-Script-Nonce'] = wdtScript.nonce;}
```

Comment:
```suggestion
                const wdt = document.querySelector('.sf-toolbar');
                if (!wdt) return;

                const wdtStyle = wdt.nextElementSibling;
                if (!wdtStyle) return;  

                if (wdtStyle.nonce) {
                    event.detail.fetchOptions.headers['X-SymfonyProfiler-Style-Nonce'] = wdtStyle.nonce;
                }

                const wdtScript = wdtStyle.nextElementSibling;
                if (!wdtScript) return;     
                           
                if (wdtScript.nonce) {
                     event.detail.fetchOptions.headers['X-SymfonyProfiler-Script-Nonce'] = wdtScript.nonce;
                 }

```

Something like that, to exit early, and leverage (not-so) recent JS constructs ?